tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -163,6 +184,7 @@ export const editTextDocument = async ({
   await Promise.all([
     page.waitForResponse((resp) => resp.status() === 204 && resp.request().method() === 'PUT'),
     page.locator(textEditorInput).fill(content),
+    page.waitForSelector(saveTextFileInEditorButton, { state: 'visible' }),
```

Comment:
if the selector is `'#text-editor-controls-save:visible'` then maybe we don't need this step? Maybe it can be directly clicked
```suggestion

```